### ISSUE TITLE
Missing documentation for existing nodes

### ISSUE THREAD

**Issue by portnov** [nortikin/sverchok#2627]
There are no corresponding `.rst` files for the following nodes:

```
vd_matrix.py
viewer_typography.py
viewer_skin.py
vd_basic_lines.py
viewer_bmesh.py
viewer_idx28.py
viewer_texture.py
viewer_curves.py
vd_draw_experimental.py
viewer_texture_lite.py
viewer_polyline.py
viewer_metaball.py
lamp_out.py
stethoscope_v28.py
text_out_mk2.py
scale_mk2.py
move_mk2.py
rotation_mk2.py
input_switch_mod.py
blenddata_to_svdata2.py
obj_edit.py
BMOperatorsMK2.py
cache.py
objects_in_lite.py
particles.py
group.py
dupli_instances_mk4.py
monad.py
curve_in.py
obj_remote_mk2.py
uv_texture.py
instancer.py
3dview_props.py
create_bvh_tree.py
particles_MK2.py
node_remote.py
mask_convert.py
join_tris.py
csg_booleanMK2.py
convex_hull_mk2.py
udp_client.py
cricket.py
lattice_analyzer.py
vertex_colors_mk3.py
sort_blenddata.py
object_raycast2.py
closest_point_on_mesh2.py
sample_uv_color.py
select_mesh_verts.py
scene_raycast2.py
sculpt_mask.py
weightsmk2.py
set_blenddata2.py
points_from_uv_to_mesh.py
custom_mesh_normals.py
getsetprop.py
get_asset_properties.py
color_uv_texture.py
armature_analyzer.py
filter_blenddata.py
matrix_array.py
scalar_mk3.py
easing.py
spiral.py
edge_split.py
multi_exec.py
sn_functor_b.py
topology_simple.py
sn_functor.py
color_in_mk1.py
normal.py
formula_deform_MK2.py
color_out_mk1.py
interpolation_mk2.py
formula_color.py
slice_lite.py
numpy_array.py
bmesh_obj_in.py
bmesh_out.py
bmesh_to_element.py
bmesh_analyzer_big.py
flip_normals.py
mesh_beautify.py
grid_fill.py
limited_dissolve.py
pulga_physics.py
extrude_multi_alt.py
limited_dissolve_mk2.py
extrude_separate_lite.py
mesh_separate_mk2.py
planar_edgenet_to_polygons.py
subdivide_lite.py
unsubdivide.py
triangulate_heavy.py
symmetrize.py
vd_attr_node.py
filter_empty_lists.py
volume.py
linked_verts.py
distance_point_plane.py
raycaster_lite.py
distance_point_line.py
distance_line_line.py
kd_tree_edges_mk2.py
deformation.py
bvh_overlap_polys.py
intersect_line_sphere.py
diameter.py
mesh_select.py
intersect_plane_plane.py
mesh_filter.py
image_components.py
bvh_nearest_new.py
edge_angles.py
normals.py
kd_tree_path.py
points_inside_mesh.py
bbox_mk2.py
distance_pp.py
kd_tree_MK2.py
path_length.py
proportional.py
area.py
object_insolation.py
select_similar.py
polygons_centers_mk3.py
```

**Comment by vicdoval** [comment]
https://github.com/nortikin/sverchok/blob/b28_prelease_master/docs/nodes/analyzers/intersect_plane_plane.rst
https://github.com/nortikin/sverchok/blob/b28_prelease_master/docs/nodes/analyzers/intersect_line_sphere.rst
https://github.com/nortikin/sverchok/blob/b28_prelease_master/docs/nodes/analyzers/deformation.rst
https://github.com/nortikin/sverchok/blob/b28_prelease_master/docs/nodes/analyzers/distance_line_line.rst
https://github.com/nortikin/sverchok/blob/b28_prelease_master/docs/nodes/analyzers/bbox_mk2.rst
https://github.com/nortikin/sverchok/blob/b28_prelease_master/docs/nodes/analyzers/distance_line_line.rst
and maybe others, they are there, what is the problem with them?
and about pulga_phisics.rst. I was in  the 2.7 branch... i missed it no i try to add it in #2622  and Travis throws a failure. The node is on the alpha category but on the modifiers_make folder. I placed the docs on the alpha folder. Did I miss anything?

**Comment by vicdoval** [comment]
Or maybe I should ignore this tests by the moment?

**Comment by portnov** [comment]
Most probable problems are
* node file name is not equal to documentation file name
* node directory is not equal to documentation directory name

**Comment by portnov** [comment]
Aha, I see, under `docs/` we have `analyzers` (note the *s*), but under `nodes/` we have `analyzer` (without *s*)... probably there are other similar problems too.

@vicdoval you may add your node to known problems list (in the `docs_tests.py`) for now if you wish.

**Comment by vicdoval** [comment]
There is just one fail in #2622 that probably is due the analyzer 's', Do you want me to change the folder name or ignore it and leave it to you?

**Comment by portnov** [comment]
I'll rename the directory.

**Comment by portnov** [comment]
Renamed in `b28_prelease_master`.

**Comment by portnov** [comment]
@zeffii what should we do with `alpha_nodes` and `beta_nodes`? For other directories, there is 1-to-1 correspondence between directories under `docs/nodes` and under `nodes`; but we do not have `nodes/alpha_nodes` or `nodes/beta_nodes`. Should we remove corresponding directories from `docs/nodes`? Or...?

**Comment by zeffii** [comment]
@portnov i propose to drop those doc/directories .

I further propose the inclusion of some short-form documentation in the `.py` of each node that doesn't have ` .rst`, maybe as a class property. Or maybe docstring extension..
```
node_doc

class SvNodeName(....):
    ...
    sv_shortform_doc = """    """  # 

```
in which several things are made clear to the user
```
1.  this node is not very well tested ( feel free to help and provide feedback at the issuetracker
2.  known issues with the node (limitations, algorithm implementation details)
3.  a link to the development thread(s) / pull request that spawn the node
4.  a mention of the creators handle, optionally.
5.  a date of node submit
6.  what else?
```
I suspect few users take the time to read the docs, altho i think we provide a most convenient way to do so with the `nodepanel / item / open documentation buttons`.  Perhaps an additional spawnable panel from the Rclick menu of nodes could show a small popup of "Node Info".


```

**Comment by portnov** [comment]
I've dropped `alpha_nodes` and `beta_nodes` under `docs`.

As for docstring... Well, ideally each node should have docstring, even those ones with documentation files — `rst` file is user documentation, docstring is more for developers. 

But, as you can see, there are many nodes without any documentation. If anyone has enough time to write either docstrings or regular documentation for them, you're welcome! :)

**Comment by zeffii** [comment]
i'll do a proof of concept for vd_experimental , maybe :)

**Comment by zeffii** [comment]
>  docstring is more for developers.

i agree, and also disagree. I am becoming a big proponent of not separating documentation that way.

the bpy py console is massively unutilized.
```
>>> nodes = bpy.data.node_groups['NodeTree'].nodes
>>> n = nodes['VD Experimental']
>>> help(n)

# 1. bla...
blablabla
# 2. blablabl
blablablablablabal

etc.

ascii schema

```
would be awesome.
```

**Comment by zeffii** [comment]
but going off topic a little bit.   i've been meaning to do some popup based documentation POC . this is good reason to do that today.

**Comment by portnov** [comment]
Technically, at sverchok loading time, we could read `rst` documentation and do `for node in nodes: node.__doc__ = read_rst_documentation(node)` :)
not sure if it would be that useful, though...

**Comment by zeffii** [comment]
perhaps overwriting __doc__ is a bit hardcore :)
```
>>> n.help   
```
would be a reasonable SvNodeCommon solution
